source/agora/consensus/EnrollmentManager.d

```diff
@@ -451,6 +483,26 @@ public class EnrollmentManager
         catch (Exception ex)
         {}
     }
+
+    /***************************************************************************
+
+        Function that reverse Hash by converting it into Byte array
+
+        Params:
+            data = hash of target to reverse
+
+        Return:
+            Reverse bytes and return to the new ubyte[]
+
+    ***************************************************************************/
+
+    private static ubyte[] reverseUbytes (ref const Hash data) @trusted nothrow
+    {
+        ubyte[] bytes;
+        foreach (val; data[].dup.reverse)
```

Comment:
Why the `dup` here ? It allocates data for no reason. If `reverse` is in place, then you can just `return data[].dup.reverse`, no need for the loop. If it does not allocate, then why the `dup` ?

Reply:
Oh, that's right. applied the suggestion.